tests/components/media_player/test_directv.py

```diff
@@ -0,0 +1,507 @@
+async def async_media_previous_track(hass, entity_id=None):
+    """Send the media player the command for prev track."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    await hass.services.async_call(DOMAIN, SERVICE_MEDIA_PREVIOUS_TRACK, data)
+
+
+async def async_play_media(hass, media_type, media_id, entity_id=None,
+                           enqueue=None):
+    """Send the media player the command for playing media."""
+    data = {ATTR_MEDIA_CONTENT_TYPE: media_type,
+            ATTR_MEDIA_CONTENT_ID: media_id}
+
+    if entity_id:
+        data[ATTR_ENTITY_ID] = entity_id
+
+    if enqueue:
+        data[ATTR_MEDIA_ENQUEUE] = enqueue
+
+    await hass.services.async_call(DOMAIN, SERVICE_PLAY_MEDIA, data)
+
+
+class MockDirectvClass:
+    """A fake DirecTV DVR device."""
+
+    def __init__(self, ip, port=8080, clientAddr='0'):
+        """Initialize the fake DirecTV device."""
+        self._host = ip
+        self._port = port
+        self._device = clientAddr
+        self._standby = True
+        self._play = False
+
+        self._locations = LOCATIONS
+
+        self.attributes = LIVE
+
+    def get_locations(self):
+        """Mock for get_locations method."""
+        test_locations = {
+            'locations': self._locations,
+            'status': {
+                'code': 200,
+                'commandResult': 0,
+                'msg': 'OK.',
+                'query': '/info/getLocations'
+            }
+        }
+
+        return test_locations
+
+    def get_standby(self):
+        """Mock for get_standby method."""
+        return self._standby
+
+    def get_tuned(self):
+        """Mock for get_tuned method."""
+        if self._play:
+            self.attributes['offset'] = self.attributes['offset']+1
+
+        test_attributes = self.attributes
+        test_attributes['status'] = {
+            "code": 200,
+            "commandResult": 0,
+            "msg": "OK.",
+            "query": "/tv/getTuned"
+        }
+        return test_attributes
+
+    def key_press(self, keypress):
+        """Mock for key_press method."""
+        if keypress == 'poweron':
+            self._standby = False
+            self._play = True
+        elif keypress == 'poweroff':
+            self._standby = True
+            self._play = False
+        elif keypress == 'play':
+            self._play = True
+        elif keypress == 'pause' or keypress == 'stop':
+            self._play = False
+
+    def tune_channel(self, source):
+        """Mock for tune_channel method."""
+        self.attributes['major'] = int(source)
+
+
+async def test_setup_platform_config(hass):
+    """Test setting up the platform from configuration."""
+    with MockDependency('DirectPy'), \
+            patch('DirectPy.DIRECTV', new=MockDirectvClass):
+
+        await async_setup_component(hass, mp.DOMAIN, WORKING_CONFIG)
+        await hass.async_block_till_done()
+
+    assert sum(1 for item in hass.data['media_player'].entities) == 1
```

Comment:
Don't assert things about entitites. Assert states for the corresponding entity_id or domain.

Reply:
What would be the best method then to check the total number of entities within a component? Idea here is to make sure that only 1 entity was created.

Reply:
Get all states for the domain.

Reply:
Well, that did not mean I knew how to do it. :-)

Found the StateMachine class in the core, able to figure it out from there. :-)

One thing for sure, also learning a lot more about the inners of HASS and how things are distributed throughout the files <g>.